sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -41,6 +41,20 @@ namespace usm {
 void *alignedAllocHost(size_t Alignment, size_t Size, const context &Ctxt,
                        alloc Kind, const property_list &PropList,
                        const detail::code_location &CodeLoc) {
+  // if no device in context has spect::usm_host_allocations then raise
+  // feature_not_supported exception
+  bool raiseUsmException = true;
+  for (auto device : Ctxt.get_devices()) {
+    if (device.has(sycl::aspect::usm_device_allocations)) {
```

Comment:
```suggestion
    if (device.has(sycl::aspect::usm_host_allocations)) {
```

Also, I think it might be slightly more concise if `std::any_of` / `std::none_of` is used, but that's a matter of taste?

Reply:
+1 for `std::none_of`; it gets rid of the extra auxiliary variable